backend/onedrive/quickxorhash/quickxorhash.go

```diff
@@ -136,7 +136,8 @@ func (q *quickXorHash) Write(p []byte) (n int, err error) {
 func (q *quickXorHash) checkSum() (h [Size]byte) {
 	// Output the data as little endian bytes
 	ph := 0
-	for _, d := range q.data[:len(q.data)-1] {
```

Comment:
This doesn't copy the `q.data` array, it slices it, so the backing store remains in the same place. Thus its quite a cheap thing to do as all it is is filling in 3 fields in the slice header.

If you look at a disassembly you'll see that your code saves about 5 instructions of setup.

https://godbolt.org/z/jh1rM7coY

So it is probably worth doing but the difference it makes will be very small.